src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/ExecutorFactory.cs

```diff
@@ -33,8 +35,10 @@ public static class ExecutorFactory
                 var arguments = new object[handler.Parameters.Length];
                 for (var i = 0; i < handler.Parameters.Length; i++)
                 {
+                    // TODO: This should be cached?
+                    var binder = page.PageContext.HttpContext.RequestServices.GetRequiredService<PageArgumentBinder>();
                     var parameter = handler.Parameters[i];
-                    arguments[i] = await page.Binder.BindModelAsync(
+                    arguments[i] = await binder.BindModelAsync(
```

Comment:
Yeah or we can just move the binding up a level. 

So the invoker calls model binding, and then calls the executor. The executor knows nothing about binding. This is the way it works for controllers. https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvoker.cs#L980

We should see if we can use `ParameterBinder` here.

It might be beneficial at this point to separate the `Binder` cleanup from the rest of this PR.